### ISSUE TITLE
"use strict" doesn't work

### ISSUE THREAD

**Issue by ayonli** [oven-sh/bun#5673]
### What version of Bun is running?

1.0

### What platform is your computer?

Darwin 22.6.0 arm64 arm

### What steps can reproduce the bug?

```js
// foo.js
"use strict";
exports.foo = function foo() {
    console.log(this);
};
```

```js
// main.js
const { foo } = require("./foo.js");

foo(); // output globalThis insteal undefined
```

### What is the expected behavior?

The correct output shall be `undefined`, which is the behavior of Node.js.

### What do you see instead?

But in Bun, the same code outputs globalThis object instead.

### Additional information

Just try:

```sh
node main.js
```

```sh
bun run main.js
```

And even if we change `main.js` to `main.mjs` and use `import`, the same issue remains.

**Comment by ayonli** [comment]
As this is a JSCore problem, for anyone encountered the same issue as I did, just replace this with new.target.

**Comment by Jarred-Sumner** [comment]
This is a transpiler bug. It should be preserving use strict I think now though

**Comment by Jarred-Sumner** [comment]
Fixed in Bun v1.1.26
